### DISCUSSION TITLE
[Windows] swanctl.exe stuck and no way to signal to stop it.

### DISCUSSION THREAD

**Discussion by psalty** [strongswan/strongswan#1553] [Q&A] [answered]
Technically it hangs if I issue the command to the connection that is already established.

.\swanctl.exe -i --ike home

**Comment by tobiasbrunner** [comment]
Please provide the daemon's complete log (see #196 for log settings, on Windows the path obviously has to be different, though).

**Comment by psalty** [comment, marked as answer]
Hi Tobias.

Based on Strongswan version 5.9.9

this is a capture from my PowerShell.
first I created an IKE SA to vcenter.

```
[IKE] maximum IKE_SA lifetime 15799s
initiate completed successfully
.\swanctl.exe -i --ike vcenter
[IKE] sending keep alive to 10.2.58.47[4500]
```

and this is the log.
```
15[CFG] vici initiate IKE_SA 'vcenter'
11[MGR] checkout IKE_SA by config
11[MGR] found existing IKE_SA 1 with config 'vcenter'
11[IKE] activating new tasks
11[IKE] nothing to initiate
11[MGR] checkin IKEv2 SA vcenter[1] with SPIs b5b053e584107353_i 681c447458a3864d_r
11[MGR] checkin of IKE_SA successful
```
So if any connection is being initiated over the existing connection swanctl just stuck and doesn't return any message like
already established connection or not.

At least in Linux, I can send SIGINT, but it doesn't work for windows build.

**Reply by tobiasbrunner** [reply]
I see. That's definitely not specific to Windows. And it's a side-effect of no event getting triggered when initiating an IKE_SA that's already established. Note that establishing IKE_SAs (i.e. calling `swanctl -i` without `-c` option) is not that common and only possible since support for childless IKE_SA was added (also requires support for it from the responder).

When initiating an IKE_SA childless, the controller waits for a state change to IKE_ESTABLISHED, but that will obviously never happen if the IKE_SA is already in that state (technically it could also be in some other state e.g. IKE_REKEYING). Something like the patch in the _1553-childless-reinit_ branch should fix this.

**Reply by psalty** [reply]
Thanks. I applied the path and it looks like working fine.
